src/components/modes/teacher/Settings.js

```diff
@@ -236,35 +320,73 @@ class Settings extends Component {
-            <FormControlLabel
-              control={programmingLanguageSelectControl}
-              label={t('Programming Language')}
-            />
+        <div hidden={tabIndex !== TABS.indexOf(DISPLAY_SETTINGS_TAB)}>
+          <Grid
+            className={classes.container}
+            container
+            direction="column"
+            spacing={2}
+            alignItems="stretch"
+          >
+            <Grid item>
+              <FormControl>
+                <FormGroup>
+                  <FormControlLabel
+                    control={topBarVisibleSwitchControl}
+                    label={t('Show Top Bar to Students')}
+                  />
+                  <FormControlLabel
+                    control={versionNavSwitchControl}
+                    label={t('Show Version Navigation')}
+                  />
+                  <FormControlLabel
+                    control={editButtonSwitchControl}
+                    label={t('Show Code Edit Button')}
+                  />
+                  <FormControlLabel
+                    control={visibilityToggleSwitchControl}
+                    label={t('Show Visibility Toggle')}
+                  />
+                  <FormControlLabel
+                    control={codeSampleVisibilitySwitchControl}
+                    label={t("Allow students to see other students' code")}
+                  />
+                </FormGroup>
+              </FormControl>
+            </Grid>
           </Grid>
-          <Grid item className={classes.editor}>
-            <FormLabel>{t('Code')}</FormLabel>
-            <Editor
-              height="50vh"
-              defaultLanguage={programmingLanguage}
-              language={programmingLanguage}
-              value={code}
-              onChange={this.handleChangeEditor('code')}
-              options={{
-                scrollBeyondLastLine: false,
-                detectIndentation: false,
-                tabSize: 2,
-              }}
-            />
+        </div>
+        <div hidden={tabIndex !== TABS.indexOf(CODE_SETTINGS_TAB)}>
+          <Grid
+            className={classes.container}
+            container
+            direction="column"
+            spacing={2}
+            alignItems="stretch"
+          >
+            <Grid item>
+              <FormControlLabel
+                control={programmingLanguageSelectControl}
+                label={t('Programming Language')}
+              />
+            </Grid>
+            <Grid item className={classes.editor}>
+              <FormLabel>{t('Code')}</FormLabel>
+              <Editor
+                height="50vh"
+                defaultLanguage={programmingLanguage}
+                language={programmingLanguage}
+                value={code}
+                onChange={this.handleChangeEditor('code')}
+                options={{
+                  scrollBeyondLastLine: false,
+                  detectIndentation: false,
+                  tabSize: 2,
```

Comment:
Not for this commit, but this should depend on the programming language. We should create a helper function for that or a map of programming language to default tab size.

Reply:
Yes that is a good idea !
Created #46 to address this.